fidesctl/src/fidesctl/cli/utils.py

```diff
@@ -28,3 +32,42 @@ def handle_cli_response(
         finally:
             sys.exit(1)
     return response
+
+
+def with_analytics(ctx: click.Context, command_handler: Callable, **kwargs: Dict) -> Any:  # type: ignore
+    """
+    Send an `AnalyticsEvent` with details about the executed command,
+    as long as the CLI has not been configured to opt out of analytics.
+
+    :param ctx: The command's execution `click.Context` object
+    :param command_handler: The handler function defining the evaluation logic for the analyzed command
+    :param **kwargs: Any arguments that must be passed to the `command_handler` function
+    """
+
+    command = " ".join(filter(None, [ctx.info_name, ctx.invoked_subcommand]))
+    error = None
+    executed_at = datetime.now(timezone.utc)
+    status_code = 0
+
+    try:
+        return command_handler(**kwargs)
+    except Exception as err:
+        error = type(err)
+        status_code = 1
+        raise err
+    finally:
+        if (
+            ctx.obj["CONFIG"].user.analytics_opt_out is False
+        ):  # requires explicit opt-in
+            event = AnalyticsEvent(
+                "CLI Command Executed",
+                executed_at,
+                command=command,
+                docker=bool(getenv("RUNNING_IN_DOCKER") == "TRUE"),
```

Comment:
We plan to let `dbt` handle the transformations on the data in Snowflake since there is quite a bit that will be happening in there (parsing of json, etc.) and it can handle that work easily